gradle-plugin/src/main/kotlin/org/mozilla/apt/ServiceLocator.kt

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.apt
+
+import com.google.common.annotations.VisibleForTesting
+import org.mozilla.apt.shell.Git
+import org.mozilla.apt.shell.GitAggregates
+
+/**
+ * A container for dependencies following the service locator pattern.
+ *
+ * We do not own the lifecycle of tasks that are constructed directly in projects that apply this
+ * plugin so we're forced to use a dependency injection pattern if we want to test. Using a DI
+ * framework in gradle sounds unnecessarily complicated (if not impossible) so we use a service
+ * locator.
+ */
+open class ServiceLocator {
+
+    open val git by lazy { Git(Runtime.getRuntime()) }
+    open val gitAggregates by lazy { GitAggregates(git) }
+
+    @VisibleForTesting protected fun overrideInstance(instance: ServiceLocator) {
```

Comment:
Nit: the default value of the `otherwise` argument for `VisibleForTesting` is `private`, but this should probably be `none`.

Reply:
Now that I'm trying to open this project again, I can't import the annotation. I don't know where it came from before!

I replaced it with a commented version: `/* @VisibleForTesting(otherwise = NONE) */`. Since this is not being worked on heavily, I think it's okay to use comments rather than importing the library.

Reply:
Turns out the macOS JDK 8 comes with this annotation but the Android Studio SDK 28 JVM does not. In both cases, neither includes the `otherwise` parameter – I think that comes from the support libraries. In any case, it's hard to guarantee people will use the macOS JDK over Android (b/c Android Studio) so I'm fine leaving it commented.